File: 812-LargestTriangleArea/LargestTriangleArea.cpp
class Solution {
public:
    double largestTriangleArea(vector<vector<int>>& points) {
        double max_area = -1;
        for (const auto& p1 : points) {
            double x1 = p1[0], y1 = p1[1];
            for (const auto& p2 : points) {
                double x2 = p2[0], y2 = p2[1];
                for (const auto& p3 : points) {
                    double x3 = p3[0], y3 = p3[1];
                    double cur_area = fabs(x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) / 2.0;
                    if (max_area < cur_area) {
                        max_area = cur_area;
                    }
                }
            }

        }
        return max_area;
    }
};

File: 812-LargestTriangleArea/LargestTriangleArea.py
class Solution(object):
    def largestTriangleArea(self, points):
        """
        :type points: List[List[int]]
        :rtype: float
        """
        max_area = -1
        for x in points:
            x1, y1 = x[0], x[1]
            for y in points:
                x2, y2 = y[0], y[1]
                for z in points:
                    x3, y3 = z[0], z[1]
                    area = abs(x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) / 2.0
                    if area > max_area:
                        max_area = area
        return max_area